### ISSUE TITLE
[ST3/4] Package Control not working on macOS

### ISSUE THREAD

**Issue by NicosKaralis** [wbond/package_control#1612]
I tried instaling the Package Control via Command Palette and even manually but it was not working, after some debugging I found that the _ctypes_ file for python3.3 are not being correctly loaded

This is the relevant part in the debug
```
Traceback (most recent call last):
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py", line 305, in <module>
  File "./python3.3/ctypes/__init__.py", line 366, in __getattr__
  File "./python3.3/ctypes/__init__.py", line 371, in __getitem__
AttributeError: dlsym(0x7f876fc44440, EVP_PKEY_size): symbol not found
```

<details><summary>Here is the rest of the output log</summary>
<p>

```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 308, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.3/importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/2_bootstrap.py", line 26, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/bootstrap.py", line 20, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/download_manager.py", line 24, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/downloaders/__init__.py", line 20, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/downloaders/oscrypto_downloader.py", line 44, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/tls.py", line 11, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_mac/tls.py", line 39, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_mac/util.py", line 208, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto.py", line 15, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py", line 688, in <module>
Package Control.package_control.deps.oscrypto._ffi.FFIEngineError: Error initializing ctypes
reloading python 3.3 plugin Package Control.Package Control
Traceback (most recent call last):
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py", line 305, in <module>
  File "./python3.3/ctypes/__init__.py", line 366, in __getattr__
  File "./python3.3/ctypes/__init__.py", line 371, in __getitem__
AttributeError: dlsym(0x7f876fc44440, EVP_PKEY_size): symbol not found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 308, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.3/importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/Package Control.py", line 89, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/commands/__init__.py", line 3, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/commands/advanced_install_package_command.py", line 10, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/package_manager.py", line 34, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/downloaders/__init__.py", line 20, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/downloaders/oscrypto_downloader.py", line 44, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/tls.py", line 11, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_mac/tls.py", line 39, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_mac/util.py", line 208, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto.py", line 15, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/Users/nicoskaralis/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py", line 688, in <module>
Package Control.package_control.deps.oscrypto._ffi.FFIEngineError: Error initializing ctypes
```
</p>
</details>

Any ideas on how to solve this?

**Comment by ioctopuses** [comment]
I have the same problem with MacOS 12.1. In the past, the old MacOS has installed sublime text, and there is no problem when upgrading to 12.1罗马字母的第23个字母

**Comment by df-a** [comment]
Same problem here as well, on MacOS Monterrey, using OpenSSL v3.0.1
```console
Traceback (most recent call last):
  File "/Users/aviftw/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py", line 305, in <module>
  File "./python3.3/ctypes/__init__.py", line 366, in __getattr__
  File "./python3.3/ctypes/__init__.py", line 371, in __getitem__
AttributeError: dlsym(0x2214322e0, EVP_PKEY_size): symbol not found
```

**Comment by rgaudin** [comment]
Indeed the symbol is missing

```sh
$ grep EVP_PKEY_size  /usr/local/lib/libcrypto.dylib
```

It is found in openssl 1.1 though (brew installed):

```sh
$ grep EVP_PKEY_size  /usr/local/opt/openssl@1.1/lib/libcrypto.dylib
Binary file /usr/local/opt/openssl@1.1/lib/libcrypto.dylib matches
```

**Comment by ashu-tosh-kumar** [comment]
> @rgaudin there's a typo in your final line (missing the `b` in `dylib`)
> 
> For others, the command that should fix this issue (at least on OS Monterey)
> 
> ```shell
> brew install openssl@1.1 && ln -sf $(set -- `brew --cellar openssl@1.1`/1.1.1? && echo "$1")/lib/libcrypto.dylib /usr/local/lib/
> 
> # or for an exact command:
> ln -sf /usr/local/Cellar/openssl@1.1/1.1.1o/lib/libcrypto.dylib /usr/local/lib/
> ```
> 
> EDIT: note that the command suggested by @rgaudin includes the path `openssl@1.1/1.1.1n` but my installation had the path `openssl@1.1/1.1.1o` instead, so I made the first version of the command above to be more portable.

Running `ln -sf /usr/local/Cellar/openssl@1.1/1.1.1o/lib/libcrypto.dylib /usr/local/lib/` solved the issue for me. Thanks.

**Comment by faelin** [comment]
> if i install sublime text 4126 using official sublimetext, will i face this issues?
> or while installing brew for the first time on mac will cause this on silicon mac?

Most likely, any installation on Big Sur or above will face this issue (has nothing to do with the processor, as I have an Intel MBP and had this issue). This issue stems from the fact that many programs still rely on older versions of openssl, and are not compatible newer versions of the library. The version of the library `libcrypto.dylib` is the important factor here, which is linked from OpenSSL, *not* something provided by Sublime. In other words, the way you install Sublime probably doesn't matter (if the app came bundled with the appropriate library, then it wouldn't matter which method you use to install it, because the regular download doesn't use an installer wizard to pull down dependencies).

**Comment by deathaxe** [comment]
Please don't quote previous comments all over the time!

That said, Package Control uses oscrypto python package for HTTPS communication, because ST2 and first releases of ST3 didn't support SSL. oscrypto relies on ssl libraries from OS as the name says.

Package Control fails loading as openssl 3 ABI changed in ways which cause import errors, so needed modules are not loaded.

Otherwise using urllib downloader would probably fix the issue as it uses ST's bundled openssl libraries these days.

**Comment by nbrooks09** [comment]
I have the same issue. I am running MacOS Monterey 12.6 and a freshly installed version of Sublime Text 3 (Build 1426). When I install Package Control, it seems to install correctly but cannot be used. Package Control is not in my "ignored packages" as in some other users' issues. When I try to go into the Package Control settings - everything is greyed out (see below)

<img width="1440" alt="Screen Shot 2022-10-11 at 2 47 12 PM" src="https://user-images.githubusercontent.com/8354247/195174629-0538e097-d5fb-4ddf-95ba-5d0bd231f6d4.png">

**Comment by faelin** [comment]
@nbrooks09 — the screenshot you've attached appears to be an unrelated problem. You have `libcrypto` installed correctly, however your installation is built for the x86 architecture used by Intel macs, while your Sublime console output says that you need the arm64 build (i.e. you have an M1 mac, and therefore need the M1-compatible libcrypto).

You may be able to solve this by replacing your Homebrew installation (if you use one) to the [ARM-based Homebrew](https://github.com/Homebrew/discussions/discussions/707#discussioncomment-342325), or by [manually building arm64 libcrypto](https://stackoverflow.com/questions/69002453/how-to-build-openssl-for-m1-and-for-intel), or you might be able to find an answer in [this openssl github issue](https://github.com/openssl/openssl/pull/12369).

If someone with an M1 mac find a solution, please let us know!

**Comment by nbrooks09** [comment]
@faelin thanks for this!

I had the ARM-based Homebrew installed already.

What worked for me was to [move libcrypto.dylib](https://stackoverflow.com/questions/65202385/openssl-libcrypto-dylib-problem-with-package-control-in-sublime-text-3).

**Comment by FichteFoll** [comment]
I have the same problem on Arch Linux now where it appears to find libcrypto.so.3 instead of the also available libcrypto.so.1.1

```
reloading python 3.3 plugin Package Control.2_bootstrap
Traceback (most recent call last):
  File "/home/mhm/.config/sublime-text/Packages/Package Control/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py", line 305, in <module>
    P_EVP_PKEY
  File "./python3.3/ctypes/__init__.py", line 366, in __getattr__
  File "./python3.3/ctypes/__init__.py", line 371, in __getitem__
AttributeError: /usr/lib/libcrypto.so.3: undefined symbol: EVP_PKEY_size

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/sublime_text/Lib/python33/sublime_plugin.py", line 308, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.3/importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/opt/sublime_text/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/2_bootstrap.py", line 26, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/opt/sublime_text/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/bootstrap.py", line 20, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/opt/sublime_text/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/download_manager.py", line 24, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/opt/sublime_text/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/downloaders/__init__.py", line 20, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/opt/sublime_text/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/downloaders/oscrypto_downloader.py", line 44, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/opt/sublime_text/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/tls.py", line 23, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/opt/sublime_text/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/tls.py", line 10, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/opt/sublime_text/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libssl.py", line 7, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/opt/sublime_text/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto.py", line 15, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/opt/sublime_text/Lib/python33/sublime_plugin.py", line 1692, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "/home/mhm/.config/sublime-text/Packages/Package Control/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py", line 688, in <module>
    raise FFIEngineError('Error initializing ctypes')
Package Control.package_control.deps.oscrypto._ffi.FFIEngineError: Error initializing ctypes
```

There is functionality in oscrypto to override library paths via a `_backend_config` dict that seems to be initialized during the `use_openssl` call, but I failed to see a configuration available to the end user to override this within Package Control.

By creating an override file that corrects the path that oscrypto tries to load libcrypto for, I am able to initialize Package Control again. 

```diff
--- Installed Packages/Package Control/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py	2020-10-02 10:34:52
+++ Packages/Package Control/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py	2022-11-28 14:47:20
@@ -23,7 +23,7 @@
 
 libcrypto_path = _backend_config().get('libcrypto_path')
 if libcrypto_path is None:
-    libcrypto_path = get_library('crypto', 'libcrypto.dylib', '42')
+    libcrypto_path = get_library('crypto.so.1', 'libcrypto.dylib', '42')
 if not libcrypto_path:
     raise LibraryNotFoundError('The library libcrypto could not be found')
 
```

Interestingly other packages that try to import things from the PC library (such as the AutomaticPackageReloader importing as `PackageManager`) fail to load and raise the exception above, even though both are inside of the 3.3 plugin host *and* the module is available:

<details>

```
reloading python 3.3 plugin AutomaticPackageReloader33.package_reloader
Traceback (most recent call last):
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py", line 305, in <module>
  File "./python3.3/ctypes/__init__.py", line 366, in __getattr__
  File "./python3.3/ctypes/__init__.py", line 371, in __getitem__
AttributeError: /usr/lib/libcrypto.so.3: undefined symbol: EVP_PKEY_size

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/sublime_text/Lib/python33/sublime_plugin.py", line 308, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.3/importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 584, in _check_name_wrapper
  File "<frozen importlib._bootstrap>", line 1022, in load_module
  File "<frozen importlib._bootstrap>", line 1003, in load_module
  File "<frozen importlib._bootstrap>", line 560, in module_for_loader_wrapper
  File "<frozen importlib._bootstrap>", line 868, in _load_module
  File "<frozen importlib._bootstrap>", line 313, in _call_with_frames_removed
  File "/home/mhm/.config/sublime-text/Packages/AutomaticPackageReloader33/package_reloader.py", line 1, in <module>
    from AutomaticPackageReloader import package_reloader as package_reloader38  # noqa
  File "/home/mhm/.config/sublime-text/Packages/AutomaticPackageReloader/package_reloader.py", line 8, in <module>
    from .reloader import reload_package
  File "/home/mhm/.config/sublime-text/Packages/AutomaticPackageReloader/reloader/__init__.py", line 1, in <module>
    from .reloader import reload_package, load_dummy
  File "/home/mhm/.config/sublime-text/Packages/AutomaticPackageReloader/reloader/reloader.py", line 13, in <module>
    from .resolver import resolve_parents
  File "/home/mhm/.config/sublime-text/Packages/AutomaticPackageReloader/reloader/resolver.py", line 2, in <module>
    from package_control.package_manager import PackageManager
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/package_manager.py", line 34, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/downloaders/__init__.py", line 20, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/downloaders/oscrypto_downloader.py", line 44, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/tls.py", line 23, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/tls.py", line 10, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libssl.py", line 7, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto.py", line 15, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py", line 688, in <module>
package_control.deps.oscrypto._ffi.FFIEngineError: Error initializing ctypes

>>> import sys
>>> {k: v for k, v in sys.modules.items() if "libcrypto" in k}
{}
Python 3.3>>> import sys
Python 3.3>>> {k: v for k, v in sys.modules.items() if "libcrypto" in k}
{'Package Control.package_control.deps.oscrypto._openssl._libcrypto_ctypes': <module 'Package Control.package_control.deps.oscrypto._openssl._libcrypto_ctypes' from '/home/mhm/.config/sublime-text/Packages/Package Control/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py'>, 'Package Control.package_control.deps.oscrypto._openssl._libcrypto': <module 'Package Control.package_control.deps.oscrypto._openssl._libcrypto' from '/home/mhm/.config/sublime-text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto.py'>}
```

</details>

**Comment by deathaxe** [comment]
I'd be curious whether it stil happens in four-point-oh branch as it ships with latest oscrypto libs.

**Comment by FichteFoll** [comment]
I can try that out some time later if I don't forget. Is there anything else I need to consider before just cloning the repo into my Packages folder?

**Comment by deathaxe** [comment]
Basically not, I hope. I've tested migration from PC3.x to PC4 several times with success on Windows and Linux (Elementary OS). It should convert the library structure (move all to Lib) and should be ready after a single restart of ST.

The only limitation so far is libraries being reinstalled rather than updated when calling Satisfy Libraries as required PEP404 and PEP508 version comparison is not yet applied.

I use that branch in production without noticible issues however.

**Comment by FichteFoll** [comment]
Quick update, since my local setup became so broken I needed to update my package and wasn't able to go around this anymore. Following are my steps and observations during the update process:

1. Close ST.
2. Removed Package Control from Installed Packages and clone into Packages (`master` branch!).
3. Start ST, packages are updated (=> the problem appears to be fixed in `master` already)
4. Close ST.
5. Checkout `four-point-oh` branch.
6. Start ST, wait for the PC popup asking me to restart, close ST.
7. Start ST, wait for the PC popup asking me to restart, close ST.
8. Start ST, no popup anymore and also almost no tracebacks in the console anymore except for this one:

```
Traceback (most recent call last):
  File "/opt/sublime_text/Lib/python33/sublime_plugin.py", line 528, in on_api_ready
    plc()
  File "/home/fichte/.config/sublime-text/Installed Packages/SublimeLinter.sublime-package/sublime_linter.py", line 54, in plugin_loaded
    package_control.events.install('SublimeLinter') or
AttributeError: 'module' object has no attribute 'events'
```

**Comment by CodeByZach** [comment]
The solution put forth by @faelin worked for me. I ended up running the following:
`ln -sf /usr/local/Cellar/openssl@1.1/1.1.1u/lib/libcrypto.dylib /usr/local/lib/`

@faelin Thank you!!

**Comment by Arbior** [comment]
On my Mac,  with Ventura, the path was slightly different (1.1.1u -> 1.1.1k).
Running ln -sf /usr/local/Cellar/openssl@1.1/1.1.1k/lib/libcrypto.dylib /usr/local/lib/ solved the problem.

PS: the problem occurred after the installation of BeautifyLatex with LatexIndent.

**Comment by inferiorhumanorgans** [comment]
> The solution put forth by @faelin worked for me. I ended up running the following: `ln -sf /usr/local/Cellar/openssl@1.1/1.1.1u/lib/libcrypto.dylib /usr/local/lib/`
> 
> @faelin Thank you!!

This is a *bad* idea as it will break anything that expects the default library to be OpenSSL 3.

**Comment by inferiorhumanorgans** [comment]
> ```json
>  "install_prereleases":
>  [
>  	"Package Control",
>  ],
> ```

Is there any way to bootstrap this?

**Comment by deathaxe** [comment]
Package Control will be uppraded next time package update is triggered, either by automatic updates at ST startup or by manually calling "Package Control: Upgrade All Packages/Overwrite All" via Command Palette.

In case PC3.x is not working, you can 
1. download https://github.com/wbond/package_control/archive/refs/tags/4.0.0-beta2.zip
2. rename it to _Package Control.sublime-package_
3. call _Main Menu > Preferences > Browse Packages...
4. navigate to to _../Installed Packages_ 
5. move _Package Control.sublime-package_ to ST's _Installed Packages_

**Comment by brian-farrell** [comment]
@deathaxe The file that gets downloaded in your instructions above is named 'Package.Control.sublime-package'

I found that I needed to **_rename_** the file to 'Package Control.sublime-package' in order for this fix to work.  (I changed the first Period, after the word 'Package' to a _space_)

Also needed to restart ST3.

Package Control appears to be working again as expected after this.

**Comment by inferiorhumanorgans** [comment]
> Package Control will be uppraded next time package update is triggered, either by automatic updates at ST startup or by manually calling "Package Control: Upgrade All Packages/Overwrite All" via Command Palette.
> 
> In case PC3.x is not working, you can
> 
>     1. download https://github.com/wbond/package_control/releases/download/4.0.0-beta2/Package.Control.sublime-package
> 
>     2. call _Main Menu > Preferences > Browse Packages..._
> 
>     3. navigate to to _../Installed Packages_
> 
>     4. move _Package Control.sublime-package_ to ST's _Installed Packages_

Right, but that only works if PC has been installed and bootstrapped.  On a clean install these steps don't work.

**Comment by deathaxe** [comment]
@brian-farrell That's unfortunatelly a restriction of GitHub download artefacts, that whitespace is automatically replaced by period.

Forgot about that.

@inferiorhumanorgans Package Control bootstraps itself. There's nothing more to do, than placing it into _Installed Packages_. Maybe a restart is needed after it was loaded the first time. but that's it. Please check file name. The download artefact contains an unwanted period.

**Comment by gerardroche** [comment]
My unit tests for osx have started failing. I assume it's related to this issue. It times out when trying to install Package Control:

```
...
/Users/runner/work/NeoVintageous/NeoVintageous
Run . $GITHUB_ACTION_PATH/../../scripts/utils.sh
copy the package to sublime package directory
/Users/runner/work/NeoVintageous/NeoVintageous
Installing UnitTesting from https://github.com/SublimeText/UnitTesting@1.5.9 to /Users/runner/Library/Application Support/Sublime Text/Packages/UnitTesting
Installing coverage from https://github.com/codexns/sublime-coverage@1.0.0 to /Users/runner/Library/Application Support/Sublime Text/Packages/coverage
Run bash $GITHUB_ACTION_PATH/../../scripts/install_package_control.sh
creating Package Control.sublime-settings
/Users/runner/work/_actions/SublimeText/UnitTesting/v1/actions/setup/../../scripts/install_package_control.sh: line 67:  1293 Terminated: 15          subl
/Users/runner/work/_actions/SublimeText/UnitTesting/v1/actions/setup/../../scripts/install_package_control.sh: line 67:  1873 Terminated: 15          subl
/Users/runner/work/_actions/SublimeText/UnitTesting/v1/actions/setup/../../scripts/install_package_control.sh: line 67:  2060 Terminated: 15          subl
....................................Timeout: Fail to install Package Control.
Error: Process completed with exit code 1.
```

https://github.com/NeoVintageous/NeoVintageous/actions/runs/5601698773

**Comment by deathaxe** [comment]
It would require PC4.0 to be released and the sublime-package file being uploaded to packagecontrol.io or Unittest to be tweaked to use PC4.0 beta builds. Whatever happens first (or ever).

**Comment by cht21** [comment]
> Package Control will be uppraded next time package update is triggered, either by automatic updates at ST startup or by manually calling "Package Control: Upgrade All Packages/Overwrite All" via Command Palette.
> 
> In case PC3.x is not working, you can
> 
>     1. download https://github.com/wbond/package_control/releases/download/4.0.0-beta2/Package.Control.sublime-package
> 
>     2. call _Main Menu > Preferences > Browse Packages..._
> 
>     3. navigate to to _../Installed Packages_
> 
>     4. move downloaded package into _Installed Packages_ and rename it to _Package Control.sublime-package_ (replace the `.` by ` `).

Thanks, this worked for me on Ventura 13.4 after a clean install of ST4. Moving the file to "Installed Packages" made the difference. There's another thread on Github saying it should be placed in "Packages", but that didn't do the trick.

**Comment by deathaxe** [comment]
PC 4.0 moves all dependencies such as _lsp_utils_ from _Data/Packages/_ to _Data/Lib/python33_ to make them available for import without needing _00_package_control_loader_ plugin and to tidy up _Data/Packages_.

Except some "plugin-like" functions of some dependencies from no longer being available, it shouldn't effect how plugins work.

In fact I am using LSP and LSP-pyright (or any other LSP-helper) without issues on python 3.3 (and with some local tweaks even on python 3.8) plugin_host.

Once dependencies have been moved, you might need to completely exit and restart ST for changes to take effect.

**Comment by DiegoAz** [comment]
@deathaxe I just installed PC 4.0 beta4, LSP and LSP-pyright and everything works as expected and it worked instantly. Thanks. 
I think I might have messed up yesterday when I modified the $PATH in ~/.zprofile

**Comment by eugconrad** [comment]
> @rgaudin there's a typo in your final line (missing the `b` in `dylib`)
> 
> For others, the command that should fix this issue (at least on OS Monterey)
> 
> ```shell
> brew install openssl@1.1 && ln -sf $(set -- `brew --cellar openssl@1.1`/1.1.1? && echo "$1")/lib/libcrypto.dylib /usr/local/lib/
> 
> # or for an exact command:
> ln -sf /usr/local/Cellar/openssl@1.1/1.1.1o/lib/libcrypto.dylib /usr/local/lib/
> ```
> 
> EDIT: note that the command suggested by @rgaudin includes the path `openssl@1.1/1.1.1n` but my installation had the path `openssl@1.1/1.1.1o` instead, so I designed the top version of the above command to be more portable.

Running `ln -sf /usr/local/Cellar/openssl@1.1/1.1.1o/lib/libcrypto.dylib /usr/local/lib/` solved the issue for me too. Thank you so much!

**Comment by Caffeinated-Python** [comment]
For anyone with still having this issue and has homebrew installed, unlinking openssl from homebrew resolved the issue for me.

**tldr** at the bottom

**Steps**
1) Close and fully quit the Sublime Text editor.
2) In a terminal, enter the following command:
`brew unlink openssl`
(on my system, takes around ~ 10 seconds to finish)
3) Open the Sublime Text editor, open then Command palette (Command + Shift + P):
Enter 'Install Package Control' and select the drop down option, you should get the text:
**"**Package Control was successfully installed
Use the Command Palette and type "Install Package" to get started**"**

4) Click ok, open the command palette again, and enter the text 'Package control'. One should be able to see the various Package Control options available (List Packages, Remove Package, Advanced Install Package, etc...)

Note - if you don't see any of the Package Control options, try fully closing and quitting Sublime Text again. You can verify Sublime Text is fully terminated by checking for the Sublime Text process to drop (disappear) in the Activity Center application. I found some instances in which I had to open and quit Sublime Text a few times before the package control options finally appear listed in the Command Palette.

5) Perform whatever package installations, removals, etc you need to to.
6) In the terminal again, enter `brew link openssl`

Caveat - the packages I've installed and used are still functional after using `brew link openssl`. There maybe certain use cases in which the steps above may not work for you.

Not exactly the most elegant approach, it still leaves Package control in a broken state but for anyone that is hesitant on using the `ln -sf` approach or downloading/moving installation files around, this workaround may be suffice until a more permanent solution is pushed up.

**System info used for test**: Fresh install of Big Sur (11.7.9) and fresh install of Sublime Text 4 (stable channel, build 4152).
Packages used: HTML-CSS-JS Prettyify, Pretty JSON, Json Colors and Navigation.

**tldr** - Fully quit sublime text, in a terminal use `brew unlink openssl`, use package control for whatever needs you have, `brew link openssl` in a terminal to return to your original state.

**Comment by aremmell** [comment]
> For me it was due to brew-installed openssl versions conflict. I had v1.1 (works) and 3 (doesn't).
> 
> ```shell
> $ ls -l
> /usr/local/lib/libcrypto.dylib -> /usr/local/opt/openssl@3/lib/libcrypto.dylib
> $ grep EVP_PKEY_size  /usr/local/lib/libcrypto.dylib
> $ grep EVP_PKEY_size /usr/local/Cellar/openssl@1.1/1.1.1n/lib/libcrypto.dylib
> Binary file /usr/local/Cellar/openssl@1.1/1.1.1n/lib/libcrypto.dylib matches
> # fixed by linking v1.1
> $ sudo ln -sf /usr/local/Cellar/openssl@1.1/1.1.1n/lib/libcrypto.dylib /usr/local/lib/libcrypto.dyli
> ```

This was still an issue in 2023–having openssl@3 installed in brew breaks package control–symlink fixes the issue.

**Comment by dhempy** [comment]
Wow...two years later and sublime still doesn't support the latest openssl? I've got a persnickety development stack for work (I know...) and I'm not inclined to hack around this problem to get my editor working as it did a week ago. 

I upgraded to build 4152 in the last week or so, and now all Package Control commands are missing from my Command Palette. "Install Package Control" is not on the Tools menu. (I can't pinpoint that it disappeared with the 4152 upgrade, but I am certain I used Package Control within the last month.) 

Are others seeing this with the 4152 update?

**Comment by deathaxe** [comment]
Package Control not being loaded is not related with ST version, but solely is related with OS's default openssl version, which has been pointed out in this issue a dozens of times - as the two solutions to that issue have been.

ST won't support openssl 3 anytime soon as backporting it into python 3.3 plugin_host is not too easy (see https://github.com/sublimehq/sublime_text/issues/5984).

**Comment by pwenzel** [comment]
> Wow...two years later and sublime still doesn't support the latest openssl? I've got a persnickety development stack for work (I know...) and I'm not inclined to hack around this problem to get my editor working as it did a week ago.
> 
> I upgraded to build 4152 in the last week or so, and now all Package Control commands are missing from my Command Palette. "Install Package Control" is not on the Tools menu. (I can't pinpoint that it disappeared with the 4152 upgrade, but I am certain I used Package Control within the last month.)
> 
> Are others seeing this with the 4152 update?

I am experiencing the same behavior with Sublime Text 4 build 4152 on Mac OS Monterey 12.6.7.

Installing openssl@1.1 with brew as noted above fixed it, at least for the moment.

**Comment by TKOxff** [comment]
>  The solution put forth by @faelin worked for me. I ended up running the following:
ln -sf /usr/local/Cellar/openssl@1.1/1.1.1u/lib/libcrypto.dylib /usr/local/lib/

```
ln -sf /opt/homebrew/Cellar/openssl@1.1/1.1.1u/lib/libcrypto.dylib /usr/local/lib/
```

It's work for me but the base path is different on Apple M2 Pro, macOS Ventura 13.5

**Comment by abjugard** [comment]
> Package Control not being loaded is not related with ST version, but solely is related with OS's default openssl version, which has been pointed out in this issue a dozens of times - as the two solutions to that issue have been.
> 
> ST won't support openssl 3 anytime soon as backporting it into python 3.3 plugin_host is not too easy (see [sublimehq/sublime_text#5984](https://github.com/sublimehq/sublime_text/issues/5984)).

Then they'll have to ship and maintain a hardened version of openssl 1.1.

**Comment by CodeByZach** [comment]
@wbond @deathaxe
This is now affecting a lot of users. Any way we can either standardize a proper patch or ship one out in the next version?
Thanks

**Comment by deathaxe** [comment]
Actually Package Control 4.0-beta already contains a fix. I already proposed installing it various posts before or at https://github.com/sublimehq/sublime_text/issues/6037#issuecomment-1674337039.

I can't however sign or upload a Package Control.sublime-package to official packagecontrol.io website, which was required for a final release artefact to be installable directly by ST's installer.

**Comment by rchl** [comment]
Don't you think that there should be some controlled release of 4.0 to only a small number of users to verify that it works as expected? I know that some people already use it by manually installing it but that's not quite the same as using the auto-update to install the 4.0 version.

I feel that releasing 4.0 hastily can cause more damage than good.

**Comment by deathaxe** [comment]
How do you want to achieve something like that?

The only mechanism to limit effected users is the `install_prereleases` key, which requires opt-in on users end. There's no way to do rollout in waves.

I am however more concerned about a release being limited in use as long as packagecontrol.io doesn't support python 3.8.

But that's off-topic here.

**Comment by rchl** [comment]
So I guess the closest to the real release would be by using `install_prereleases` but with package control 4 actually deployed to the servers instead of having to download it manually. Then there could be a call on Discord for more people to test it. There can only be one chance of upgrading from v3 to v4 so it's probably best to test that once v4 is 99.9% ready rather than testing early and then later beta version potentially breaking something in the upgrade process.

**Comment by deathaxe** [comment]
All 4.0.0-beta builds have been available this way. Not sure what you want more.

**Comment by faelin** [comment]
I just installed the 4.0 release manually, after my current machine ran into the lib-crypto / openssl issue.

Install worked perfectly on Mac OS Ventura (13.5), no further tweaking of libraries required!

**Comment by benfrain** [comment]
Unsure if this is the same issue. Having installed v4, I get this in the console:

```
Package Control: Installing 16 libraries...
Package Control: The library "wcwidth" is not available for Python 3.3
Package Control: The library "pathlib" is not available for Python 3.3
Package Control: The library "python-markdown" is not available for Python 3.3
Package Control: The library "backrefs" is not available for Python 3.3
Package Control: The library "ptyprocess" is not available for Python 3.3
Package Control: The library "pymdownx" is not available for Python 3.3
Package Control: The library "sublime_lib" is not available for Python 3.3
Package Control: The library "python-jinja2" is not available for Python 3.3
Package Control: The library "bracex" is not available for Python 3.3
Package Control: The library "pyte" is not available for Python 3.3
Package Control: The library "pyyaml" is not available for Python 3.3
Package Control: The library "pygments" is not available for Python 3.3
Package Control: The library "lsp_utils" is not available for Python 3.3
Package Control: The library "markupsafe" is not available for Python 3.3
Package Control: The library "mdpopups" is not available for Python 3.3
Package Control: The library "wcmatch" is not available for Python 3.3
```

**Comment by deathaxe** [comment]
That's a separate issue, related to PC 4.0 not being able to resolve compatible libraries for your version of ST, platform and CPU architecture. Could you please file a separate issue for that, to avoid cluttering this discussion with off-topics?

**Comment by benfrain** [comment]
> That's a separate issue, related to PC 4.0 not being able to resolve compatible libraries for your version of ST, platform and CPU architecture. Could you please file a separate issue for that, to avoid cluttering this discussion with off-topics?

sure: https://github.com/wbond/package_control/issues/1651 thanks

**Comment by hteumeuleu** [comment]
> ```
> ln -sf /opt/homebrew/Cellar/openssl@1.1/1.1.1u/lib/libcrypto.dylib /usr/local/lib/
> ```
> 
> It works for me too, but the base path is different on Apple M2 Pro, macOS Ventura 13.5

This one worked for me on macOS Ventura 13.5.1. Thank you!

**Comment by rsms** [comment]
> Package Control will be uppraded next time package update is triggered, either by automatic updates at ST startup or by manually calling "Package Control: Upgrade All Packages/Overwrite All" via Command Palette.
> 
> In case PC3.x is not working, you can
> 
> 1. download https://github.com/wbond/package_control/releases/download/4.0.0-beta2/Package.Control.sublime-package
> 2. call _Main Menu > Preferences > Browse Packages..._
> 3. navigate to to _../Installed Packages_
> 4. move downloaded package into _Installed Packages_ and rename it to _Package Control.sublime-package_ (replace the `.` by ` `).

This worked for me

**Comment by pohongying** [comment]
> For me it was due to brew-installed openssl versions conflict. I had v1.1 (works) and 3 (doesn't).
> 
> ```shell
> $ ls -l
> /usr/local/lib/libcrypto.dylib -> /usr/local/opt/openssl@3/lib/libcrypto.dylib
> $ grep EVP_PKEY_size  /usr/local/lib/libcrypto.dylib
> $ grep EVP_PKEY_size /usr/local/Cellar/openssl@1.1/1.1.1n/lib/libcrypto.dylib
> Binary file /usr/local/Cellar/openssl@1.1/1.1.1n/lib/libcrypto.dylib matches
> # fixed by linking v1.1
> $ sudo ln -sf /usr/local/Cellar/openssl@1.1/1.1.1n/lib/libcrypto.dylib /usr/local/lib/libcrypto.dyli
> ```

**Comment by pohongying** [comment]
yes ,I have same situation with you and i  quickly solve this problems make me a headache  when finally i delete another openssl.Thank you!

**Comment by kow-mingte** [comment]
1) **Reinstall openssl@1.1**
```shell
brew install openssl@1.1
``` 
2) **If you need to have openssl@1.1 first in your PATH, run:**
```shell
echo 'export PATH="/usr/local/opt/openssl@1.1/bin:$PATH"' >> ~/.zshrc
``` 
3) **For compilers to find openssl@1.1 you may need to set:**
  
```shell
export LDFLAGS="-L/usr/local/opt/openssl@1.1/lib"
export CPPFLAGS="-I/usr/local/opt/openssl@1.1/include"
``` 
4) **run:**
```shell
ln -sf /usr/local/Cellar/openssl@1.1/1.1.1v/lib/libcrypto.dylib /usr/local/lib/
```
5) **🎉**

**Comment by pappu687** [comment]
For me `brew unlink openssl` worked. Using ST4, MacOS Ventura.

**Comment by canidam** [comment]
Are there really no other solutions than downgrading openssl default version for the whole system?

Is there a way to do it only for Sublime?

**Comment by faelin** [comment]
### FOR ANYONE STILL STRUGGLING WITH THIS PROBLEM

Maintainer @deathaxe has explained _several times_ in this thread that you can solve the problem simply by downloading the [latest edition](https://github.com/wbond/package_control/releases/download/4.0.0-beta2/Package.Control.sublime-package) (4.0+) of Package Control!

For reference:
https://github.com/wbond/package_control/issues/1612#issuecomment-1625595350
https://github.com/wbond/package_control/issues/1612#issuecomment-1636034709
https://github.com/wbond/package_control/issues/1612#issuecomment-1675766635

### DO NOT UNINSTALL/RELINK OPENSSL

Changing the PATH variable, installed version, linked version, or otherwise modifying the version of OpenSSL used by the system is a BAD IDEA. It was always a stop-gap measure to work around the constraints of conflicting dependencies. Please do not change which OpenSSL you use unless you are ABSOLUTELY CERTAIN that you know what you are doing!

**Comment by canidam** [comment]
Thank you @faelin !
I got it working after reading this https://github.com/wbond/package_control/issues/1612#issuecomment-1636034709

I just had to download the latest package, which is https://github.com/wbond/package_control/releases/tag/4.0.0-beta8
Then place it manually in the `Installed Packages` directory. Works :)

**Comment by MatthiasPortzel** [comment]
The `ln ...` syslink solution is, at best, temporary, as it breaks when Homebrew updates OpenSSL (which is why I am back here today).

The first instructions posted by @deathaxe aren't helpful, as people experiencing this bug can't open package control settings inside Sublime.

The best (and only real) solution is in this comment (back up profile first): https://github.com/wbond/package_control/issues/1612#issuecomment-1636034709.
After following those instructions I get a popup about dependencies being migrated to Python libraries. Wait for that to finish, then restart.

**Comment by jiajianjie** [comment]
> @nbrooks09 — the screenshot you've attached appears to be an unrelated problem. You have `libcrypto` installed correctly, however your installation is built for the x86 architecture used by Intel macs, while your Sublime console output says that you need the arm64 build (i.e. you have an M1 mac, and therefore need the M1-compatible libcrypto).
> 
> You may be able to solve this by replacing your Homebrew installation (if you use one) to the [ARM-based Homebrew](https://github.com/Homebrew/discussions/discussions/707#discussioncomment-342325), or by [manually building arm64 libcrypto](https://stackoverflow.com/questions/69002453/how-to-build-openssl-for-m1-and-for-intel), or you might be able to find an answer in [this openssl github issue](https://github.com/openssl/openssl/pull/12369).
> 
> If someone with an M1 mac find a solution, please let us know!
``
my mac's cpu is arm , need reinstall brewhome in /opt

```
brew install openssl@1.1 && ln -sf $(set -- `brew --cellar openssl@1.1`/1.1.1? && echo "$1")/lib/libcrypto.dylib /usr/local/lib/\n
ll /opt/homebrew/Cellar/openssl@1.1/1.1.1w/lib/libcrypto.dylib
lrwxr-xr-x  1 jiajianjie  admin  19  9 11 22:08 /opt/homebrew/Cellar/openssl@1.1/1.1.1w/lib/libcrypto.dylib -> libcrypto.1.1.dylib

ln -sf /opt/homebrew/Cellar/openssl@1.1/1.1.1w/lib/libcrypto.1.1.dylib /usr/local/lib/libcrypto.dylib
```

**Comment by deathaxe** [comment]
Please stop spamming this issue with all those repetidive stupid posts, which already exist dozens and hundreds of times!

**Comment by frenziedcode** [comment]
> @rgaudin there's a typo in your final line (missing the `b` in `dylib`)
> 
> For others, the command that should fix this issue (at least on OS Monterey)
> 
> ```shell
> brew install openssl@1.1 && ln -sf $(set -- `brew --cellar openssl@1.1`/1.1.1? && echo "$1")/lib/libcrypto.dylib /usr/local/lib/
> 
> # or for an exact command:
> ln -sf /usr/local/Cellar/openssl@1.1/1.1.1o/lib/libcrypto.dylib /usr/local/lib/
> ```
> 
> EDIT: note that the command suggested by @rgaudin includes the path `openssl@1.1/1.1.1n` but my installation had the path `openssl@1.1/1.1.1o` instead, so I designed the top version of the above command to be more portable.

Works for me. Thank you!

**Comment by h0jeZvgoxFepBQ2C** [comment]
> Package Control will be uppraded next time package update is triggered, either by automatic updates at ST startup or by manually calling "Package Control: Upgrade All Packages/Overwrite All" via Command Palette.
> 
> In case PC3.x is not working, you can
> 
>     1. download https://github.com/wbond/package_control/releases/download/4.0.0-beta2/Package.Control.sublime-package
> 
>     2. call _Main Menu > Preferences > Browse Packages..._
> 
>     3. navigate to to _../Installed Packages_
> 
>     4. move downloaded package into _Installed Packages_ and rename it to _Package Control.sublime-package_ (replace the `.` by ` `).

There is a new release available: https://github.com/wbond/package_control/releases/download/4.0.0-beta8/Package.Control.sublime-package

**Comment by pronebird** [comment]
Time to drop Sublime into the bucket I guess...

```
~/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py", line 688, in <module>
Package Control.package_control.deps.oscrypto._ffi.FFIEngineError: Error initializing ctypes
```

**Comment by pronebird** [comment]
Time to drop Sublime into the bucket I guess...

```
~/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py", line 688, in <module>
Package Control.package_control.deps.oscrypto._ffi.FFIEngineError: Error initializing ctypes
```

**Comment by h0jeZvgoxFepBQ2C** [comment]
> Thank you @faelin ! I got it working after reading this [#1612 (comment)](https://github.com/wbond/package_control/issues/1612#issuecomment-1636034709)
> 
> I just had to download the latest package, which is https://github.com/wbond/package_control/releases/tag/4.0.0-beta8 Then place it manually in the `Installed Packages` directory. Works :)

Did you follow this? This works for most people, including me.

**Comment by pronebird** [comment]
> > Thank you @faelin ! I got it working after reading this [#1612 (comment)](https://github.com/wbond/package_control/issues/1612#issuecomment-1636034709)
> > I just had to download the latest package, which is https://github.com/wbond/package_control/releases/tag/4.0.0-beta8 Then place it manually in the `Installed Packages` directory. Works :)
> 
> Did you follow this? This works for most people, including me.

Yeah. Sadly it didn't help. I have both openssl1.1 and openssl3 installed on my mac.

**Comment by captainkovalsky** [comment]
Hey everyone,

Tried to install the latest sublime package manager manually

Got this error
```
File "/Users/vd/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py", line 688, in <module>
Package Control.package_control.deps.oscrypto._ffi.FFIEngineError: Error initializing ctypes
plugins loaded
```

https://github.com/wbond/package_control/releases/download/4.0.0-beta2/Package.Control.sublime-package

ST 4, build 4152

**Comment by rr0ss0rr** [comment]
What steps did you do to install manually?  Also beta8 is the latest.  For the latest, Check out [releases](https://github.com/wbond/package_control/releases/)

**Comment by NazarK** [comment]
> > > Thank you @faelin ! I got it working after reading this [#1612 (comment)](https://github.com/wbond/package_control/issues/1612#issuecomment-1636034709)
> > > I just had to download the latest package, which is https://github.com/wbond/package_control/releases/tag/4.0.0-beta8 Then place it manually in the `Installed Packages` directory. Works :)
> > 
> > 
> > Did you follow this? This works for most people, including me.
> 
> Yeah. Sadly it didn't help. I have both openssl1.1 and openssl3 installed on my mac.

just run:`brew link openssl@1.1` and restart sublime, it fixed it for me

**Comment by hyrious** [comment]
I have installed Package Control v4.0.0-beta8 on my macOS 14 with Sublime Text 4 with OpenSSL 3. It works perfectly.

```console
$ openssl version
OpenSSL 3.1.3 19 Sep 2023 (Library: OpenSSL 3.1.3 19 Sep 2023)

$ subl -v
Sublime Text Build 4152

$ sw_vers
ProductName:		macOS
ProductVersion:		14.0
BuildVersion:		23A344
```

If you are stuck at the place where Package Control 3.0 cannot run to update itself, you can do either of the 2 steps:

- Follow the author's instruction to manually install this package: https://github.com/wbond/package_control/issues/1612#issuecomment-1636034709
- Follow the others' in the issue that you temporarily link OpenSSL 1.0 to the system level, then your Package Control 3.0 should just work. Then go edit the `install_prereleases` section in Package Control's settings to let it update itself: https://github.com/wbond/package_control/issues/1612#issuecomment-1625595350
  - When it's done, it would be better to link OpenSSL 3 back and restart Sublime Text.

**Comment by Hoatzin** [comment]
So I was getting the _plugin_host-3.3 has exited unexpectedly_ error. Manually going through plugins and disabling them didn't do anything, so I archived my Sublime Text folder in the Application Support, and it opened without errors. I told it to install the Package Control automatically, and it did, but upon restarting the error came back. So I manually downloaded the beta, but when I started that one it said _plugin_host-3.8 has exited unexpectedly_...

I tried to force openssl 1.1 via brew but that didn't seem to alter anything or allow me to interact with Package Control menus, to sort or permute lines, to shrink or enlarge text and so on.

**Comment by roubachof** [comment]
Don't forget to rename `Package.Control.sublime-package` to `Package Control.sublime-package` (whitespace between package and control) or you will end up with a message saying it cannot find the 'Package' package in console view.

**Comment by legendbb** [comment]
I just got my issue resolved by deleting "0_package_control_loader.sublime-package" and "Package Control.sublime-package" from the Install Packages folder.
And reinstall from cmd + ctrl + p > install package control.
macOS 13.5.2, sublime 4152

**Comment by jama211** [comment]
Why has this not been fixed within the program? Why do I have to download the latest version from here, name it correctly (by removing the dot between Package.Control) instead of it working inside sublime? It's been months...

**Comment by deathaxe** [comment]
There are 2 major reasons:

1. The fix is part of the next major release 4.0, which mainly targets dropping ST2/python2 support and adding python 3.8 support. Some fine tuning is still needed to fully support dependencies on python 3.8 plugin_host. Actually, mainly a "channel" with valid metadata is the main stopper, next to support for installing python packages like pip does. But that's maybe a part for a future release anyway.
2. Only Will is able to sign a package and upload it to packagecontrol.io, so it can directly be installed from ST's main menu. It currently seems his priorities shifted.

Until then, the only chance to ship the package for download is via Github releases, which do NOT support spaces in file names. They are replaced by periods. That's why you need to rename it.

It would be easy to file a final 4.0.0 version, soon, but would still leave those behind whose systems refuses to run PC 3.x

**Comment by dak1m** [comment]
> > Running `ln -sf /usr/local/Cellar/openssl@1.1/1.1.1o/lib/libcrypto.dylib /usr/local/lib/` solved the issue for me. Thanks.
> 
> if i install sublime text 4126 using official sublimetext, will i face this issues? or while installing brew for the first time on mac will cause this on silicon mac? i'm glad that u found a fix!



> libcrypto

good to me

**Comment by plocket** [comment]
I wonder if it'd be useful to edit the original question to add the currently helpful answer(s?) to it so it (they?) don't get lost in the rest of the comments until various fixes come through. Just not sure how else to help people find the info. Took me a while to dig through here to find the two pieces that [worked](https://github.com/wbond/package_control/issues/1612#issuecomment-1636034709) for [me](https://github.com/wbond/package_control/issues/1612#issuecomment-1636476267). Thank you for the help!

**Comment by plocket** [comment]
Where should we check to find out if the related package file is signed and uploaded to packagecontrol.io, @deathaxe? That way, Future People will know where to look to find out if they can start installing via the main menu again.

**Comment by deathaxe** [comment]
The link at https://packagecontrol.io/installation would point to the PC 4.0 binary, I guess.

The official release, which is also downloaded by ST's install helper is located at https://packagecontrol.io/Package%20Control.sublime-package

Package Control 3.4.1 Metadata

Name: Package Control.sublime-package
Größe: 471460 Bytes (460 KiB)
SHA256: 817937144c34c84c88cd43b85318b2656f9c3fac02f8f72cbc18360b2c26d139

Package Control 4.0.0 would differ.

I don't want to post the details to avoid confusion with possible future releases.

**Comment by rchl** [comment]
Well, after installing 3.4.1 from the main menu, it will still update itself to 4.0 on next ST start (or sometime later), doesn't it? So effectively it's gonna be installed already anyway.

**Comment by deathaxe** [comment]
But this won't help those - especially MacOS users, who have already run into Package Control not working issue due to openssl3 being linked.

Those won't benefit from auto-update from 3.4.1 to 4.0. They must take the hard way and install manually from this repo's releases page.

**Comment by leocxy** [comment]
> Fixed by 4.0
> 
> Note: Whether Package Control 4.0 can be installed directly via Sublime Text's main menu depends on whether the related package file will be signed and uploaded to packagecontrol.io by @wbond.
> 
> Until then you may still need to
> 
>     1. download latest release from https://github.com/wbond/package_control/releases
> 
>     2. rename it to `Package Control.sublime-package`
> 
>     3. place it to ST's _Installed Packages_ directory
> 
> 
> as the beta builds before.

It work for me.
I am using a Macbook with an M2 Chip.

I downloaded the sublime 4 from the official site (Build 4169)
Then I tried to install the Package Control. It shows a successful message, but I can't see it.

If you close the ST totally and re-open it, you can see some error messages from the console. (View -> Show Console)

Now you can open the installed packages folder. 
1. Settings -> Browser packages. This will bring you to the "Packages" folder, the "Installed Packages" folder sitting at the SAME level of the "Packages" folder.
2. Go to the "Installed Packages" folder, and delete the "Package Control.sublime-package" binary file
3. Download the new release from the above release link
4. Rename and move it back to the "Installed Packages" folder.

**Comment by plocket** [comment]
> But this won't help those - especially MacOS users, who have already run into Package Control not working issue due to openssl3 being linked.

Just checking if I understand correctly:
1. This isn't going to be the situation forever. This is fixed already, the fix is just not released yet until package control switches to 4.0.
2. There's no way to check what version package control is using before downloading it, but after you download it you can check its meta data (not sure where to find that) to see what version it's using. If it's using the wrong version, you go through the procedures outlined above.

**Comment by Luka2552** [comment]
> > Fixed by 4.0
> > Note: Whether Package Control 4.0 can be installed directly via Sublime Text's main menu depends on whether the related package file will be signed and uploaded to packagecontrol.io by @wbond.
> > Until then you may still need to
> > ```
> > 1. download latest release from https://github.com/wbond/package_control/releases
> > 
> > 2. rename it to `Package Control.sublime-package`
> > 
> > 3. place it to ST's _Installed Packages_ directory
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > as the beta builds before.
> 
> It work for me. I am using a Macbook with an M2 Chip.
> 
> I downloaded the sublime 4 from the official site (Build 4169) Then I tried to install the Package Control. It shows a successful message, but I can't see it.
> 
> If you close the ST totally and re-open it, you can see some error messages from the console. (View -> Show Console)
> 
> Now you can open the installed packages folder.
> 
>     1. Settings -> Browser packages. This will bring you to the "Packages" folder, the "Installed Packages" folder sitting at the SAME level of the "Packages" folder.
> 
>     2. Go to the "Installed Packages" folder, and delete the "Package Control.sublime-package" binary file
> 
>     3. Download the new release from the above release link
> 
>     4. Rename and move it back to the "Installed Packages" folder.

Thank you very much! This worked for me! 

Sublime Text 4169

macOS Big Sur 11.6

**Comment by KvMadan** [comment]
Thank you! 
Above solution worked for 
Mac 14.2
ST 4169

**Comment by raffeyang** [comment]
Doesn't work on intel macOS.

**Comment by leonson** [comment]
I just used the solution for my Intel Macbook and unblocked. I am using Sublime Text 4 Stable Channel, Build 4169. My processor is 2.3 GHz Quad-Core Intel Core i75.

The [installation page](https://packagecontrol.io/installation)'s Manual section step 3 does not have the most recent release, thus manually downloading there did not help. The solution here has the [right link](https://github.com/wbond/package_control/releases), I would suggest either update the link on installation page's manual session, or if there is a reason not to do so, add a link(on installation page) to this issue.

In my console log I found following error message which is equivalent to this issue's:

```
Traceback (most recent call last):
  File "/Users/leonson/Library/Application Support/Sublime Text/Installed Packages/Package Control.sublime-package/package_control/deps/oscrypto/_openssl/_libcrypto_ctypes.py", line 305, in <module>
  File "./python3.3/ctypes/__init__.py", line 366, in __getattr__
  File "./python3.3/ctypes/__init__.py", line 371, in __getitem__
AttributeError: dlsym(0x7ff919620a40, EVP_PKEY_size): symbol not found
```

**Comment by deathaxe** [comment]
The original author - quite inactive recently - is the only one with write access to packagecontrol.io and thus the only one who could upload latest release there, which would also be required for ST's Package Control installer to directly install it.
